Paxdma driver enable

raveenp opened the pull request:
This PR contains changes to enable PAX DMA driver,
 - boards: arm: bcm958402m2_m7: Enable PAXDMA driver
 - boards: arm: bcm958402m2_a72: Enable PAXDMA driver

raveenp committed:
boards: arm: bcm958402m2_a72: Enable PAXDMA driver

Add DMA_IPROC_PAX_V2 config to enable PAXDMA driver.

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>

raveenp committed:
boards: arm: bcm958402m2_m7: Enable PAXDMA driver

Add DMA and DMA_IPROC_PAX_V2 config to enable PAXDMA driver.

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>

galak commented on boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig:
```diff
@@ -25,3 +25,9 @@ CONFIG_UART_CONSOLE=y
+CONFIG_DMA=y
+CONFIG_DMA_64BIT=y
+CONFIG_DMA_IPROC_PAX_V2=y
+CONFIG_DMA_0_NAME="pax_dma"
```
Remove CONFIG_DMA_0_NAME

galak commented on boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:
```diff
@@ -18,3 +18,9 @@ CONFIG_UART_NS16550=y
+CONFIG_DMA=y
+CONFIG_DMA_64BIT=y
+CONFIG_DMA_IPROC_PAX_V2=y
+CONFIG_DMA_0_NAME="pax_dma"
```
remove CONFIG_DMA_0_NAME

raveenp replied:
@galak: dts/arm/broadcom/viper-common.dtsi   contains paxdma node definition as,

paxdma: paxdma@4e100800 {
                compatible = "brcm,iproc-pax-dma-v2";
                label = "pax_dma";
                .................................
};

So CONFIG_DMA_0_NAME is set in config file to "pax_dma", otherwise I need to use "DMA_0" for pl330 and "DMA_1" for paxdma.

Let me push change in dts for setting label as "DMA_0" for pl330 and "DMA_1" for PAXDMA.

That way we can eliminate setting the names explicitly.
I think naming

galak replied:
Sounds good.

galak reviewed (changes requested):
remove CONFIG_DMA_0_NAME its not needed

galak commented on boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:
```diff
@@ -18,3 +18,9 @@ CONFIG_UART_NS16550=y
+
+# Enable PAXDMA
+CONFIG_DMA=y
+CONFIG_DMA_64BIT=y
```
Should probably also remove DMA_64BIT as its not used by the PAXDMA driver.

raveenp replied:
DMA_64BIT is needed to enable passing 64 bit source and destination buffer to the DMA driver as in our use case we need 64bit addresses.

galak replied:
gotcha.

galak commented:
WIth the iproc change merged, if you rebase this I think it should pass the CI tests.

raveenp commented:
> @galak: With the iproc change merged, if you rebase this I think it should pass the CI tests.

@galak : Yes. let me rebase.

raveenp committed:
dts: arm: broadcom: viper: rename paxdma label

To keep the generic DMA device name as defined
in the dma framework config, renamed paxdma device
label to "DMA_1".

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>

galak commented:
@raveenp looks like you still need to rebase on newer upstream to resolve the merge conflict.  You also still need to remove CONFIG_DMA_0_NAME.  There is now no code in three that uses this symbol.

raveenp commented:
> @raveenp looks like you still need to rebase on newer upstream to resolve the merge conflict. You also still need to remove CONFIG_DMA_0_NAME. There is now no code in tree that uses this symbol.

@galak: yes.  I missed to remove CONFIG_DMA_0_NAME.   Will fix and rebase on newer upstream version. Thanks.

galak commented:
@raveenp looks like we need one more rebase.

raveenp commented:
@galak : I have rebased once again.  Build may still fail due to dependency on https://github.com/zephyrproject-rtos/zephyr/pull/32690.

raveenp committed:
dts: arm: broadcom: viper-a72: paxdma: Add interrupt property

Add interrupt parent and number property specific to a72
viper platform.

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>

raveenp commented:
@galak: I have fixed current build issue by adding missing changes in a72 specific dts file with adding:
dts: arm: broadcom: viper-a72: paxdma: Add interrupt property

Still the build will fail due to dependency on
https://github.com/zephyrproject-rtos/zephyr/pull/32690: soc: arm: bcm_vk: viper: Increase default NUM_IRQS

galak commented on boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig:
```diff
@@ -32,3 +32,7 @@ CONFIG_DMA=y
 CONFIG_DMA_PL330=y
+
+# Enable PAXDMA
+CONFIG_DMA_64BIT=y
```
You are setting CONFIG_DMA_64BIT=y twice.

galak commented on boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:
```diff
@@ -30,3 +30,7 @@ CONFIG_DMA=y
 CONFIG_DMA_PL330=y
+
+# Enable PAXDMA
+CONFIG_DMA_64BIT=y
```
You are setting CONFIG_DMA_64BIT=y twice.

galak commented:
@raveenp #32690 is merged now, so should be good to rebase and fixup the last issue :)
